modules/machineconfig-garbage-collect.adoc

```diff
@@ -0,0 +1,20 @@
+// Module included in the following assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="machineconfig-garbage-collect_{context}"]
+= Managing unused rendered machine configs
+
+The Machine Config Operator (MCO) does not perform any garbage collection activities. This means that all rendered machine configs remain in the cluster. Each time a user or controller applies a new machine config, the MCO creates new rendered configs for each affected machine config pool. Over time, this can lead to a large number of rendered machine configs, which can make working with machine configs confusing. Having too many rendered machine configs can also contribute to disk space issues and performance issues with etcd. 
+
+You can remove old, unused rendered machine configs by using the `oc adm prune renderedmachineconfigs` command with the `--confirm` flag. This command allows you to remove all unused rendered machine configs or only those in a specific machine config pool. You can also remove a specific number of unused rendered machine configs in order to leave some older machine configs, in case you want to review older configurations.
+
+You can use the `oc adm prune renderedmachineconfigs` without the `--confirm` flag to see which rendered machine configs would be removed. Rendered machine configs with `as it's currently in use` in the command output would not be removed.
+
+Use the `list` subcommand to view all the rendered machine configs in the cluster or a specific machine config pool.
+
+[NOTE]
+====
+The `oc adm prune renderedmachineconfigs` command deletes only rendered machine configs that are not in use. If any of the rendered machine configs are in use by a machine config pool, the rendered machine config is not deleted, even if you specify parameters include that rendered machine config. In this case, the command output indicates why the rendered machine config was not deleted.
+====
```

Comment:
```suggestion
The Machine Config Operator (MCO) does not perform any garbage collection activities. This means that all rendered machine configs remain in the cluster. Each time a user or controller applies a new machine config, the MCO creates new rendered configs for each affected machine config pool. Over time, this can lead to a large number of rendered machine configs, which can make working with machine configs confusing. Too many rendered machine configs can also lead to disk space issues and performance issues with etcd.

You can remove old, unused rendered machine configs by using the `oc adm prune renderedmachineconfigs` command with the `--confirm` flag. With this command you can remove all unused rendered machine configs or only those in a specific machine config pool. You can also remove a certain number of unused rendered machine configs in order to keep some older machine configs, in case you want to check older configurations.

You can use the `oc adm prune renderedmachineconfigs` command without the `--confirm` flag to see which rendered machine configs would be removed. Rendered machine configs marked with `as it's currently in use` in the command output are not removed.

Use the `list` subcommand to display all the rendered machine configs in the cluster or in a specific machine config pool.

[NOTE]
====
The `oc adm prune renderedmachineconfigs` command only deletes rendered machine configs that are not in use. If a machine config pool uses one of the rendered machine configs, the rendered machine config is not deleted, even if you specify parameters that contain this rendered machine config. In this case, the command output specifies why the rendered machine config was not deleted.
====
```